subprojects/snapshots/src/main/java/org/gradle/internal/fingerprint/impl/AbstractFingerprintingStrategy.java

```diff
@@ -37,4 +58,25 @@ public String getIdentifier() {
     public CurrentFileCollectionFingerprint getEmptyFingerprint() {
         return emptyFingerprint;
     }
+
+    @Nullable
+    protected HashCode getNormalizedContentHash(FileSystemLocationSnapshot snapshot, ResourceHasher normalizedContentHasher) {
+        try {
+            return lineEndingSensitivity.isCandidate(snapshot) ?
+                normalizedContentHasher.hash(from((RegularFileSnapshot) snapshot)) :
+                snapshot.getHash();
```

Comment:
If we didn't need this `if` here, then `isCandidate()` could move to the implementation of the hasher itself, too.

Reply:
This was actually an artifact of using the pass-through hasher.  It was to avoid calling the intermediate caching hasher when we weren't going to get actually use it.  There was a slight performance penalty to doing so.  In any event, avoiding this altogether when we use default line ending sensitivity removes the need for this extra check.